### PR TITLE
Convert StringInreplaceExtension test to spec.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/test/inreplace_spec.rb [hunk 1]
```diff
@@ -0,0 +1,157 @@
+require "extend/string"
+require "tempfile"
+require "utils/inreplace"
+
+describe StringInreplaceExtension do
+  subject { string.extend(described_class) }
+
+  describe "#change_make_var!" do
+    context "flag" do
+      let(:string) { "OTHER=def\nFLAG = abc\nFLAG2=abc" }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/inreplace_spec.rb]: Use `EOS.undent` here

File: Library/Homebrew/test/inreplace_spec.rb [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+      it "is successfully replaced" do
+        subject.change_make_var! "FLAG", "def"
+        expect(subject).to eq("OTHER=def\nFLAG=def\nFLAG2=abc")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/inreplace_spec.rb]: Use `EOS.undent` here

File: Library/Homebrew/test/inreplace_spec.rb [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      end
+
+      it "is successfully appended" do
+        subject.change_make_var! "FLAG", "\\1 def"
+        expect(subject).to eq("OTHER=def\nFLAG=abc def\nFLAG2=abc")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/inreplace_spec.rb]: Use `EOS.undent` here

File: Library/Homebrew/test/inreplace_spec.rb [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      end
+
+      context "with tabs" do
+        let(:string) { "CFLAGS\t=\t-Wall -O2\nLDFLAGS\t=\t-lcrypto -lssl" }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/inreplace_spec.rb]: Use `EOS.undent` here

File: Library/Homebrew/test/inreplace_spec.rb [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+      expect(subject).to eq("fee")
+    end
+  end
+end
+
+describe Utils::Inreplace do
+  let(:file) { Tempfile.new("test") }
+
+  before(:each) do
+    file.write "a\nb\nc\n"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/inreplace_spec.rb]: Use `EOS.undent` here

File: Library/Homebrew/test/inreplace_spec.rb [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+      end
+    end
+  end
+
+  describe "#get_make_var" do
+    let(:string) { "CFLAGS = -Wall -O2\nLDFLAGS = -lcrypto -lssl" }
+
+    it "extracts the value for a given variable" do
+      expect(subject.get_make_var("CFLAGS")).to eq("-Wall -O2")
+    end
```
Comment (alyssais) [inline Library/Homebrew/test/inreplace_spec.rb]: This should be in a context like the one below

File: Library/Homebrew/.rubocop.yml [hunk 7]
```diff
@@ -12,7 +12,7 @@ AllCops:
 Style/BlockDelimiters:
   Exclude:
     - '**/cask/spec/**/*'
-    - '**/cask/test/**/*'
+    - '**/*_spec.rb'
```
Comment (MikeMcQuaid) [inline Library/Homebrew/.rubocop.yml]: Can you remind us why this is needed again?

File: Library/Homebrew/test/inreplace_spec.rb [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+    file.write <<-EOS.undent
+      a
+      b
+      c
+    EOS
+  end
+
+  after(:each) do
+    file.unlink
+  end
```
Comment (alyssais) [inline Library/Homebrew/test/inreplace_spec.rb]: Could this be one line?

Comment (reitermarkus) [inline Library/Homebrew/test/inreplace_spec.rb]: I'd like to always use `do … end` for `before` and `after` hooks, since they *do* something.

Comment (alyssais) [inline Library/Homebrew/test/inreplace_spec.rb]: They do, but usually what they're doing is side-effects of cleanup, which don't really need to draw the readers eye. For that reason, I'd prefer one-line where possible. What do you think?

**Pull request merged**